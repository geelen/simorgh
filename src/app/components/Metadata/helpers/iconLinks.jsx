import React from 'react';

export const getIconAssetUrl = (service, size) =>
  `https://news.files.bbci.co.uk/include/articles/public/${service}/images/icons/icon-${size}.png`;

const createIconLinks = (service, iconSizes, iconType) => {
  return iconSizes.map(size => {
    const iconAssetUrl = getIconAssetUrl(service, size);
    const key = `${service}-${size}`;

    if (iconType === 'icon') {
      return (
        <link
          key={key}
          rel="icon"
          type="image/png"
          href={iconAssetUrl}
          sizes={size}
        />
      );
    }

    return (
      <link key={key} rel="apple-touch-icon" sizes={size} href={iconAssetUrl} />
    );
  });
};

export const getIconLinks = (service, iconSizes) => {
  if (!iconSizes) {
    return null;
  }
  const iconTypes = Object.keys(iconSizes);
  return iconTypes.map(iconType => {
    return createIconLinks(service, iconSizes[iconType], iconType);
  });
};
