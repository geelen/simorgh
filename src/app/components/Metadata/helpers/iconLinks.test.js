import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { getIconLinks } from './iconLinks';

const iconSizes = {
  'apple-touch-icon': [
    '72x72',
    '96x96',
    '128x128',
    '144x144',
    '152x152',
    '192x192',
    '384x384',
    '512x512',
  ],
  icon: ['72x72', '96x96', '192x192'],
};

describe('getIconLinks', () => {
  const links = getIconLinks('news', iconSizes);
  shouldMatchSnapshot('should render icon meta tags correctly', links);
});
