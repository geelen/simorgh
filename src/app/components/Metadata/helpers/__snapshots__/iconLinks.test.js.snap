// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIconLinks should render icon meta tags correctly 1`] = `
<div>
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-128x128.png"
    rel="apple-touch-icon"
    sizes="128x128"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-152x152.png"
    rel="apple-touch-icon"
    sizes="152x152"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-72x72.png"
    rel="icon"
    sizes="72x72"
    type="image/png"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-96x96.png"
    rel="icon"
    sizes="96x96"
    type="image/png"
  />
  <link
    href="https://news.files.bbci.co.uk/include/articles/public/news/images/icons/icon-192x192.png"
    rel="icon"
    sizes="192x192"
    type="image/png"
  />
</div>
`;
